python/modem-firmware-1.3+/nrf_cloud_provision.py

```diff
@@ -197,21 +204,47 @@ def read_prov_csv(csv_filepath):
         csvfile.seek(0)
         prov = csv.reader(csvfile, delimiter=',')
 
-        for row in prov:
+        for row_idx, row in enumerate(prov):
             # First column in each row is the device ID
             # Add a list to the list [ <device_id>, <result_string> ]
-            device_list.append([row[0], ''])
-
-        csvfile.close()
+            try:
+                device_list.append([row[0], ''])
+            except IndexError:
+                print("Error reading row {} of provisioning CSV file.".format(row_idx + 1   ))
 
     return device_list
 
-def save_or_print(results, result_filepath):
+def read_mfwv_csv(csv_filepath):
+    mfw_ver_list = []
+    with open(csv_filepath) as csvfile:
+        mfwv = csv.reader(csvfile, delimiter=',')
+
+        for row_idx, row in enumerate(mfwv):
+            # First column in each row is the device ID
+            # Add a list to the list [ <device_id>, <mfw_version> ]
+            try:
+                mfw_ver_list.append([row[0], row[1]])
+            except IndexError:
+                print("Error reading row {} of modem firmware CSV file.".format(row_idx + 1))
+
+    return mfw_ver_list
+
+def save_or_print(results, result_filepath, append):
     # Save to file or print to console
     if len(result_filepath):
-        write_file(os.path.dirname(result_filepath),
-                   os.path.basename(result_filepath),
-                   results.getvalue().encode('utf-8'))
+        res_bytes = results.getvalue().encode('utf-8')
+        if append and os.path.exists(result_filepath):
+            try:
+                f = open(result_filepath, "ab")
```

Comment:
Consider using `with open(...):`  here, to avoid potentially failing to close the file in event of exception